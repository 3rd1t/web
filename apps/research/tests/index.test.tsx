import React from "react"
import renderer from "react-test-renderer"
import Index, { getStaticProps } from "../pages/index"

describe("Index", () => {
  it("renders correctly", () => {
    const codeExamples = [
      {
        language: "python",
        snippet: "hello snippet",
      },
      {
        language: "r",
        snippet: "hello snippet",
      },
    ]

    const tree = renderer.create(<Index codeExamples={codeExamples} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe("getStaticProps()", () => {
  it("loads the files and returns them as props", async () => {
    const res = await getStaticProps()

    expect(res.props).toBeTruthy()
    expect(res.props.codeExamples.length > 0).toBe(true)
    res.props.codeExamples.forEach((c) => {
      expect(c.language.length > 0).toBe(true)
      expect(c.snippet.length > 0).toBe(true)
    })
  })
})
