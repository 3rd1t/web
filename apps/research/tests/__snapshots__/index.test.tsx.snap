// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<div
  className="min-h-screen"
>
  <div
    className="flex flex-col items-start px-4 mt-10 space-y-8 sm:mt-12 sm:px-6 md:mt-16 lg:mt-20 lg:px-8 xl:mt-28 xl:flex-row xl:space-y-0 xl:space-x-8"
  >
    <div
      className="flex flex-col items-center w-full"
    >
      <div
        className="space-y-4 text-left sm:text-center md:space-y-8 xl:space-y-12 lg:text-left"
      >
        <h2
          className="text-4xl font-extrabold leading-10 tracking-tight text-gray-900 xl:text-5xl sm:leading-none md:text-5xl"
        >
          Get the right factor returns from our API
        </h2>
        <div
          className="text-gray-600 sm:text-lg sm:mx-auto md:text-xl lg:mx-0"
        >
          Factor investing is a strategy that chooses securities on attributes that are associated with higher returns. There are two main types of factors that have driven returns of stocks, bonds, and other factors: macroeconomic factors and style factors. The former captures broad risks across asset classes while the latter aims to explain returns and risks within asset classes.
        </div>
        <div
          className="space-y-4 md:space-x-4 md:space-y-0"
        >
          <div
            className="sm:flex sm:justify-center lg:justify-start"
          >
            <div
              className="rounded-md shadow"
            >
              <a
                className="border border-transparent rounded-sm md:py-4 md:text-lg md:px-10 flex items-center justify-center w-full px-8 py-3 text-base font-medium leading-6 transition duration-150 ease-in-out focus:outline-none text-white bg-research-900 hover:bg-research-800 hover:text-research-100"
                href="#"
              >
                Get started
              </a>
            </div>
            <div
              className="mt-3 sm:mt-0 sm:ml-3"
            >
              <a
                className="border border-transparent rounded-sm md:py-4 md:text-lg md:px-10 flex items-center justify-center w-full px-8 py-3 text-base font-medium leading-6 transition duration-150 ease-in-out focus:outline-none text-research-700 bg-research-100 hover:text-research-600 hover:bg-white"
                href="Log in"
              >
                Log in
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-auto"
    >
      <div
        className="shadow-lg"
        style={Object {}}
      >
        <nav
          className="flex items-center justify-start px-4 space-x-4 border border-gray-500 rounded-t"
          style={Object {}}
        >
          <button
            className="py-3 mr-8 text-xs font-semibold tracking-wide no-underline uppercase focus:outline-none hover:text-gray-900 text-gray-900 "
            data-testid="selectLanguage"
            onClick={[Function]}
            style={Object {}}
          >
            python
          </button>
          <button
            className="py-3 mr-8 text-xs font-semibold tracking-wide no-underline uppercase focus:outline-none hover:text-gray-900 text-gray-600"
            data-testid="selectLanguage"
            onClick={[Function]}
            style={Object {}}
          >
            r
          </button>
        </nav>
        <div
          className="px-4 overflow-x-visible overflow-y-hidden bg-gray-900 rounded-b"
          style={Object {}}
        >
          <pre
            className="font-mono"
            data-testid="syntaxHighlighter"
            style={
              Object {
                "MozHyphens": "none",
                "MozTabSize": "4",
                "OTabSize": "4",
                "WebkitHyphens": "none",
                "color": "#CBD5E0",
                "direction": "ltr",
                "fontSize": "14px",
                "hyphens": "none",
                "lineHeight": "1.375",
                "margin": ".5em 0",
                "msHyphens": "none",
                "overflow": "auto",
                "padding": "1em",
                "tabSize": "4",
                "textAlign": "left",
                "whiteSpace": "pre",
                "wordBreak": "normal",
                "wordSpacing": "normal",
              }
            }
          >
            <code
              style={
                Object {
                  "MozHyphens": "none",
                  "MozTabSize": "4",
                  "OTabSize": "4",
                  "WebkitHyphens": "none",
                  "color": "#CBD5E0",
                  "direction": "ltr",
                  "fontSize": "14px",
                  "hyphens": "none",
                  "lineHeight": "1.375",
                  "msHyphens": "none",
                  "tabSize": "4",
                  "textAlign": "left",
                  "whiteSpace": "pre",
                  "wordBreak": "normal",
                  "wordSpacing": "normal",
                }
              }
            >
              hello snippet
            </code>
          </pre>
        </div>
      </div>
    </div>
  </div>
</div>
`;
